src/lib/BulkEmailChecker.js

```diff
@@ -0,0 +1,240 @@
+/* flow */
+/* eslint-disable no-console, no-param-reassign */
+
+import dns from 'dns';
+import { Socket } from 'net';
+import EventEmitter from 'events';
+import winston from 'winston';
+import promisify from './promisify';
+
+type MxRecord = {
+    exchange: string;
+    priority: number;
+}
+
+export default class BulkEmailChecker extends EventEmitter {
+	constructor ({ maxRcptPerConn = 350, timeoutLimit = 10 * 1000, fromEmail = 'billgates@gmail' } = {}) {
+		super();
```

Comment:
Yes, you have to make the call, else it results in **no-this-befor-super** error.